### PR TITLE
fix(asff): handle empty Recommendation Url 

### LINKED ISSUE #2492: [Bug]: Unable to push findings to Security Hub when Recommendation Url is blank.
### Steps to Reproduce

We're running prowler against aws, reporting into Security Hub using asff.  Any similar setup should produce this same outcome, as long as all default checks are used. 

The first check in my output I see with this failure is `prowler-backup_reportplans_exist`.  Security Hub rejects this finding due to the Remediation Recommendation Url being blank.

### Expected behavior

All Prowler findings are successfully reported to Security Hub.

ALSO, it's expected that Security Hub will accept a finding with a blank Url, per their documentation: https://docs.aws.amazon.com/securityhub/1.0/APIReference/API_Recommendation.html

Manually adding a URL to the finding gets Security Hub to accept the finding, so their documentation is likely incorrect.

### Actual Result with Screenshots or Logs

Sample finding:

```json
{
    "SchemaVersion": "2018-10-08",
    "Id": "prowler-backup_reportplans_exist-[FILTERED}",
    "ProductArn": "arn:aws:securityhub:us-east-1::product/prowler/prowler",
    "RecordState": "ACTIVE",
    "ProductFields": {
        "ProviderName": "Prowler",
        "ProviderVersion": "3.6.0",
        "ProwlerResourceName": "arn:aws:iam::[FILTERED]:root"
    },
    "GeneratorId": "prowler-backup_reportplans_exist",
    "AwsAccountId": "[FILTERED]",
    "Types": [
        "Recover",
        "Resilience",
        "Backup"
    ],
    "FirstObservedAt": "2023-06-14T16:54:10Z",
    "UpdatedAt": "2023-06-14T16:54:10Z",
    "CreatedAt": "2023-06-14T16:54:10Z",
    "Severity": {
        "Label": "LOW"
    },
    "Title": "Ensure that there is at least one AWS Backup report plan",
    "Description": "No Backup Report Plan Exist",
    "Resources": [
        {
            "Type": "Other",
            "Id": "arn:aws:iam::[FILTERED]:root",
            "Partition": "aws",
            "Region": "us-east-1"
        }
    ],
    "Compliance": {
        "Status": "FAILED",
        "RelatedRequirements": [],
        "AssociatedStandards": []
    },
    "Remediation": {
        "Recommendation": {
            "Text": "Use AWS Backup to create backup report plans that provide visibility into the success or failure of backup operations.",
            "Url": ""
        }
    }
}
```

Sample log output:
```2023-06-14 16:58:02,313 [File: security_hub.py:50] 	[Module: security_hub]	 ERROR: Failed to send archived findings to AWS Security Hub -- InvalidInput -- Finding does not adhere to Amazon Finding Format. data.Remediation.Recommendation.Url should NOT be shorter than 1 characters, data.Remediation.Recommendation.Url should match format "uri".```

### How did you install Prowler?

From pip package (pip install prowler)

### Environment Resource

Fargate Task (on ECS)

### OS used

Wolfi via Chainguard docker images

### Prowler version

Prowler 3.6.0

### Pip version

pip 23.1.2

### Context

Populating the Url anywhere it is not populated may do the trick, there may be good URLs for checks that are missing URLs, or `https://` may be enough?

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Handle empty Recommendation Url since Security Hub does not accept them empty.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/outputs/json.py [hunk 1]
```diff
@@ -78,6 +78,9 @@ def fill_json_asff(finding_output, audit_info, finding, output_options):
         AssociatedStandards=associated_standards,
         RelatedRequirements=compliance_summary,
     )
+    # Fill Recommendation Url if it is blank
+    if not finding.check_metadata.Remediation.Recommendation.Url:
```
Comment (n4ch04) [inline prowler/lib/outputs/json.py:84]: Could you create a test for this case please?

Comment (MrCloudSec) [reply prowler/lib/outputs/json.py:84]: We cannot create it since moto does not cover this call and we cannot mock the call twice.

Comment (jfagoagas) [reply prowler/lib/outputs/json.py:84]: This has nothing to do with Moto, is just a test for the `fill_asff_json`. I'm finishing it and I'll push it later today.

Comment (MrCloudSec) [reply prowler/lib/outputs/json.py:84]: Thanks 😄

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**